Readme.md

```diff
@@ -386,6 +388,174 @@ export function MyForm({ isNewItem, addItem, updateItem }: Props): JSX.Element =
+            v.target.value
+          )
+        }
+      />
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+---
+
+## FieldSet
+
+A FieldSet is basically an array of FormFields. Contrary to nested objects, a FieldSet will only be instantiated if it is **explicitely** declared in the model.
+
+This is because at the moment of field instantiation, the addField method within a Form (or a FormField if it is a nested object) will as if the model value is actually an Array. It is not possible of implicitely determine if the field is an array based only on the type.
+
+The FieldSet class extends Array, so you can access all methods an array has, plus custom methods that work similarly as those in FormField (more in the [API](#fieldset-1))
+
+```ts
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {
+      emails: ['example@email.com'],
+    },
+    handleSubmit: async () => {
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      // a FieldSet is iterable
+      {fields.emails.map((field, index) => {
+        // Every FieldSet item is a FormField
+        return (
+          <div className="input">
+            <input
+              value={field.value}
+              onChange={(v) => field.onChange(v.target.value)}
+            />
+            // FieldSet has a helpful removeField method
+            <button onClick={() => fields.emails.remove(index)}>remove</button>
+          </div>
+        );
+      })}
+      // FieldSet also has an addFields method to create new items
+      <button onClick={() => fields.emails.addFields('')}>Add Field</button>
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+### Adding Items
+
+The `addFields` methods takes an array of items that can be either a simple value or an object with value and validation. Being an array extension, it is also possible to add items to FieldSet using array methods like `push`;
+
+```ts
+// simple value
+<button onClick={() => fields.emails.addFields('example@email.com')}>Add Field</button>
+// value with validation
+<button onClick={() => fields.emails.addFields({
+  value: '',
+  validation: 'email'
+})}>Add Field</button>
+// Multiple Fields
+<button onClick={() => fields.emails.addFields('example@email.com', '', 'another-example@emal.com')}>Add Field</button>
+```
+
+**Important**
+
+`addFields` takes a comma separated array of parameters, if you would like to pass an array you will need to destructure it
+
+```ts
+const newValues = ['example@email.com', '', 'another-example@emal.com'];
+<button onClick={() => fields.emails.addFields(...newValues)}
+```
+
+### Removing Items
+
+There is a `removeField` helper function that takes in an index number as parameter. Under the hood it is a simple call to `splice`. You can also use array methods like `pop` or `shift` to remove items.
+
+```ts
+<button onClick={() => fields.emails.remove(index)}>remove</button>
+```
+
+---
```

Comment:
What I'm missing here is the main use case of arrays: arrays of objects.
Assuming I have a model like 
```ts
interface Model {
  images: Array<{
    id: string;
    tags: string[];
    file?: Blob;
  }>
}
```

I want to access it like
```ts
form.fields.images.fields[0].tags.valid;
form.fields.images.fields[0].file.required;
form.fields.images.add({id: "1", tags: []});
```